lib/Bio/EnsEMBL/DataCheck/Checks/CompareStableIdsMapping.pm

```diff
@@ -0,0 +1,90 @@
+  GROUPS         => ['compare_core'],
+  DATACHECK_TYPE => 'critical',
+  DB_TYPES       => ['core'],
+  TABLES         => ['meta', 'stable_id_event', 'mapping_session', 'gene', 'transcript', 'exon', 'translation'],
+};
+
+sub tests {
+  my ($self) = @_;
+
+  SKIP: {
+    my $old_dba = $self->get_old_dba();
+
+    skip 'No old version of database', 1 unless defined $old_dba;
+
+    my $new_dba = $self->dba;
+    my $stable_id_prefix = $new_dba->get_MetaContainerAdaptor->single_value_by_key('species.stable_id_prefix');
+    my $old_stable_id_prefix = $old_dba->get_MetaContainerAdaptor->single_value_by_key('species.stable_id_prefix');
+
+    #If the stable ids are different, which should probably not happen we do not check the stable id
+    #mapping as it is not needed
+    if ($stable_id_prefix ne $old_stable_id_prefix) {
+      skip 'Different stable id prefix'.$new_dba->dbc->dbname.' and '.$old_dba->dbc->dbname, 1;
+    }
+
+    #If the gene set has changed, we need to make sure that the stable id mapping has been done.
+    #If the count for each table is the same as previous release, it is highly unlikely that the
+    #gene set has changed
+    my $table_num_count = 0;
+    my $object_count_sql = 'SELECT COUNT(*) FROM %s';
+    my @tables = qw(gene transcript exon translation);
+    foreach my $table (@tables) {
+      my $old_count = $old_dba->dbc->sql_helper()->execute_single_result( -SQL => sprintf($object_count_sql, $table) );
+      my $new_count = $new_dba->dbc->sql_helper()->execute_single_result( -SQL => sprintf($object_count_sql, $table) );
+      ++$table_num_count if ($old_count == $new_count);
+    }
+    my $expected_value = 1;
+    if ($table_num_count == scalar(@tables)) {
+      $expected_value = undef;
+    }
+    else {
+      my $old_session_sql = 'SELECT old_db_name, old_release FROM mapping_session WHERE old_db_name = "'.$old_dba->dbc->dbname.'"';
+      is_rows($new_dba, $old_session_sql, 1, 'Checking '.$old_dba->dbc->dbname.' is present in mapping_session as old_db_name');
+
+      my $new_session_sql = 'SELECT new_db_name, new_release FROM mapping_session WHERE new_db_name = "'.$new_dba->dbc->dbname.'"';
+      is_rows($new_dba, $new_session_sql, 1, 'Checking '.$new_dba->dbc->dbname.' is present in mapping_session as new_db_name');
+    }
+
+    #If the gene set has changed we expect the count to be higher than the previous database, hence minimum value is 1.
+    #Otherwise we expect the row count to be equal.
+    #In the unlikely event of extending exon and causing a change of version in the gene, transcript or translation, the
+    #datacheck will either erroneously fail or succeed.
+    my $stable_id_event_sql = 'SELECT old_stable_id, new_stable_id FROM stable_id_event';
+    my $stable_id_event_desc = 'Check that the stable_id_event table has as many or more rows than the previous database';
+    row_totals($new_dba, $new_dba, $stable_id_event_sql, undef, $expected_value, $stable_id_event_desc);
```

Comment:
As you commented above, it should connect to old_dba and new_dba.
I used undef as the code will use sql1 if sql2 is not defined but I can use stable_id_event_sql twice.
My understanding of the row_totals method is if param 5 is undef, it does a perfect match and if param 5 is set the result has to be equal or higher. If the gene set has changed, there should be more rows in the new database. But if it hasn't, I should not have more rows in the new database so I would want the test to fail.
I will change it to 1 because I'm looking at some Havana stuff and they just change coordinates without adding new objects...

Reply:
Your interpretation of row_totals is correct - I didn't get that the test should fail if there were unexpectedly more rows in the new database.